src/util/peakPicking.ts

```diff
@@ -17,23 +15,50 @@ export function peakPicking(
     xVariable = 'x',
     yVariable = 'y',
     optimize = false,
-    max = true,
+    expectedWidth = 1,
+    max: isMax = true,
+    shapeOptions = {},
   } = options;
 
   const x = spectrum.variables[xVariable]?.data;
-  const y = spectrum.variables[yVariable]?.data;
+  let y;
+  if (!isMax) {
+    y = spectrum.variables[yVariable]?.data.slice(); // do deep copy as we maybe need to flip sign
+  } else {
+    y = spectrum.variables[yVariable]?.data;
+  }
+
   if (!x || !y) return;
   let targetIndex;
+  targetIndex = xFindClosestIndex(x, target);
+  let optimizedPeak;
+  let optimizedIndex;
+  const result: Record<string, number> = {};
   if (optimize) {
-    targetIndex = max
-      ? xyMaxClosestYPoint({ x, y }, { target }).index
-      : xyMinClosestYPoint({ x, y }, { target }).index;
+    if (isMax === false) {
+      let maximumY = max(y);
+      for (let i = 0; i < y.length; i++) {
+        y[i] *= -1;
+        y[i] += maximumY; // This makes it somewhat more robust
+      }
+    }
+
+    optimizedPeak = optimizePeak(
+      { x, y },
+      [{ x: x[targetIndex], y: y[targetIndex], width: expectedWidth }],
+      shapeOptions,
+    );
+
+    optimizedIndex = xFindClosestIndex(x, optimizedPeak.peaks[0].x);
+    for (let key in spectrum.variables) {
+      result[key] = spectrum.variables[key].data[optimizedIndex];
+    }
+    result.width = optimizedPeak.peaks[0].width;
```

Comment:
@kjappelbaum So we don't store the optimized values ?

For the 2 optimized values of xVariable and yVariable I would put the optimized values
```
result[xVariable] = optimizedPeak.peaks[0].x;
result[yVariable] = optimizedPeak.peaks[0].y;
```

Reply:
right, currently we use an x/y that is actually there and closest to the optimized position. 

```
    optimizedIndex = xFindClosestIndex(x, optimizedPeak.peaks[0].x);
    for (let key in spectrum.variables) {
      result[key] = spectrum.variables[key].data[optimizedIndex];
    }
```

but I can also use the ones the optimizer spits out, we might even make a separate flag for this.